src/sentry/api/bases/organization.py

```diff
@@ -294,6 +297,12 @@ def has_release_permission(self, request, organization, release):
         Does the given request have permission to access this release, based
         on the projects to which the release is attached?
         """
-        return ReleaseProject.objects.filter(
-            release=release, project__in=self.get_projects(request, organization)
-        ).exists()
+        key = "release_perms:1:%s" % hash_values([request.user.id, organization.id, release.id])
```

Comment:
The only thing I’m not sure about that I’d like verified, how does this work with API keys? request isn’t always a user.

Reply:
Good point. I'll look at the get_projects further to see how it handles it

Reply:
We'll need a way to coalesce `request.user.id` and `ApiKey.id` down to just get a unique identifier of the authn mechanism.